### PR TITLE
DAOS-14015 test: Updating the targets_str config val.

### PR DESCRIPTION
**Pull request by rpadma2**
Test-tag: test_daos_server_config_basic
Test-repeat: 3

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/server/daos_server_config.yaml [hunk 1]
```diff
@@ -167,5 +167,5 @@ server_config_val: !mux
   targets_str:
     config_val:
       - "targets"
-      - "ABDCD"
+      - -1
```
Comment (phender) [inline src/tests/ftest/server/daos_server_config.yaml:170]: We want both fail test cases, `-1` and `ABDCD`, so this should be a complete new entry.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/server/daos_server_config.yaml (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/server/daos_server_config.yaml:170]: This change is already done with the latest patch and is undergoing CI testing.

File: src/tests/ftest/server/daos_server_config.yaml [hunk 2]
```diff
@@ -169,3 +169,8 @@ server_config_val: !mux
       - "targets"
       - "ABDCD"
       - "FAIL"
+  targets_str_negative:
```
Comment (phender) [inline src/tests/ftest/server/daos_server_config.yaml:172]: Minor nit, but this was previously called `targets_negative`

**Review by phender — DISMISSED** [review]

File: src/tests/ftest/server/daos_server_config.yaml (hunk 2 above)
Comment (rpadma2) [reply src/tests/ftest/server/daos_server_config.yaml]: done.

**Comment by rpadma2** [comment]
All tests passed.

**Review by shimizukko — APPROVED** [review]

**Comment by rpadma2** [comment]
@daos-stack/daos-gatekeeper : This PR can be merged.

**Comment by daltonbohning** [comment]
> @daos-stack/daos-gatekeeper : This PR can be merged.

@phender has requested changes, so we need to wait for @phender to dismiss or approve. But it looks like they were addressed

**Pull request merged**